packages/react/src/hooks/useMinter.ts

```diff
@@ -0,0 +1,54 @@
+import { useWallet } from '../WalletContext';
+import { callContractMethod, MAX_GAS, mint } from '@mintbase-js/sdk';
+import { useState } from 'react';
+import { MintArgs } from '@mintbase-js/sdk/lib/mb-impl/token.types';
+
+export type MinterHookReturn = {
+  loading: boolean;
+  error: any;
+  data: any;
+  methods: any;
+};
+
+export type MinterHookArgs = MintArgs;
+
+export const useMinter = (args: MintArgs): MinterHookReturn => {
+  const { wallet, activeAccountId } = useWallet();
+  const { nftContractId, options } = args;
+
+  const [loading, setLoading] = useState(false);
+
+  const handleMint = async (): Promise<void> => {
+    setLoading(true);
+
+    // TODO: use @mintbase/storage to get a reference
+    const reference = '';
+    
+    await callContractMethod(
```

Comment:
do we want to borrow @tifrel 's `execute` name?  I don't love `callContractMethod` but it is rather explicit

Reply:
I would actually be a fan of `view` and `exec` (or `call`, but I find that too generic) to signal a difference between mutations and pure information retrieval

Reply:
I like execute personally.  I had thought to make a separate @mintbase-js/rpc package for view calls but maybe we can keep in SDK

Reply:
Created a generic `execute` method that takes objects of the type `NearContractCall` or an array of `NearContractCall`